torch/nn/modules/flatten.py

```diff
@@ -57,8 +57,8 @@ class Unflatten(Module):
       (tuple of `(name, size)` tuples) for `NamedTensor` input.
 
     Shape:
-        - Input: :math:`(N, *dims)`
-        - Output: :math:`(N, C_{\text{out}}, H_{\text{out}}, W_{\text{out}})`
+        - Input: :math:`(*)`. Input can be any shape.
+        - Output: :math:`(*, *unflattened\_size, *)`.
```

Comment:
Any way to indicate that `*unflattened_size` is expanded at `dim`? Maybe something like `(*, dim, *)` before and `(*, *unflattened\_size, *)` after (where `*` represents any number of dimensions, including none)- wdyt?

Reply:
`(*, dim, *)` would be inconsistent with how the size is normally specified in other Shape descriptions.

What do you think of using `dim` as a subscript: `(*, S_{dim}, *)`, where `S_{dim}` is the "size at dimension `dim`".

Reply:
Cool, I agree with you that using `dim` as a subscript is more accurate. We could even explicitly point out that `S_{dim} = \prod *unflattened\_size` must hold - thoughts on that?